craft_parts/plugins/base.py

```diff
@@ -36,6 +36,9 @@
     from craft_parts import infos
 
 
+PackageFileList = dict[tuple[str, str], set[pathlib.Path]]
```

Comment:
I'm wondering whether we're restricting ourselves too much here with a tuple with exactly two values. For example, what happens if we decide to add more metadata (say, origin url)? 

Maybe we should use a dataclass instead?

Reply:
Yeah that's a good point and shouldn't be a big change, I'll do that.

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/1997a6f7d3344d07c6beb7993f95ce064486894d